app/src/components/pointcloud/PointCloud.tsx

```diff
@@ -310,6 +325,13 @@ const Projection = React.memo(function Projection() {
     });
   }, [referenceData, eventIdToGroup, pointGroupVisibility]);
 
+  const filteredCorpusData = useMemo(() => {
+    return corpusData.filter((point) => {
+      const group = eventIdToGroup[point.eventId];
+      return pointGroupVisibility[group];
+    });
+  }, [corpusData, eventIdToGroup, pointGroupVisibility]);
```

Comment:
keeping it for now just so we can have this PR merged, since it's needed for the visibility control and i don't know what i'm doing here and am getting kind of lost